src/compiler.rs

```diff
@@ -241,10 +265,27 @@ macro_rules! push_frame {
             $compiler.frames.push(Frame::new());
 
             // Return value stored as a variable
-            $compiler.define_variable("", Type::Unknown, &mut $block).unwrap();
+            let var = Variable::new("", true, Type::Unknown);
+            $compiler.define(var).unwrap();
+
             $code
 
-            $compiler.frames.pop().unwrap();
+            let frame = $compiler.frames.pop().unwrap();
+            // 0-th slot is the function itself.
+            for var in frame.stack.iter().skip(1) {
+                if !(var.read || var.upvalue) {
+                    let e = ErrorKind::SyntaxError(
+                        var.line,
+                        Token::Identifier(var.name.clone()
+                    ));
+                    $compiler.error_on_line(
+                        e,
+                        var.line,
+                        Some(format!("Usage of undefined value: '{}'.", var.name))
+                    );
+                }
+                $compiler.panic = false;
```

Comment:
Maybe we'd like a proper warning!-macro instead of resetting the panic. Then it'd be easier to output "5 warnings found" or something.

Reply:
Out of scope though.